src/main/java/org/wildfly/security/auth/server/MechanismConfiguration.java

```diff
@@ -41,14 +43,19 @@
  * @author <a href="mailto:david.lloyd@redhat.com">David M. Lloyd</a>
  */
 public final class MechanismConfiguration {
+
+    private final MechanismConfiguration parent;
     private final NameRewriter preRealmRewriter;
     private final NameRewriter postRealmRewriter;
     private final NameRewriter finalRewriter;
     private final RealmMapper realmMapper;
     private final Map<String, MechanismRealmConfiguration> mechanismRealms;
     private final List<SecurityFactory<Credential>> serverCredentialFactories;
+    private final Map<String, MechanismConfiguration> serverSpecificConfiguration;
```

Comment:
I think "host" or "hostName" would be a better term than "server" here, but in terms of common usage it seems to be split with protocols such as HTTP and DNS preferring "host" and SNI and SASL preferring "server".